### PR TITLE
refactor(tabset): implement design tokens

### PR DESCRIPTION
**Pull request by lalithkarikelli**
## Description

`<hx-tabset>`: refactor with design tokens

### What are the relevant story cards/tickets? Any additional PRs or other references?

Jira: SURF 2205

## Before you request a review for this PR:

- [x] For UI changes, did you manually test in recent versions of modern browsers (Chrome, Firefox, and Safari)?
- [x] For UI changes, did you manually test in IE11 and legacy Edge?
- [ ] Did you add component tests for any new code?
- [ ] Did you run the component unit tests via `yarn test` to ensure all tests passed?
- [ ] Did you include a screenshot of the component tests?
- [ ] If you changed/added functionality, did you update the demo page and documentation?
- [ ] If needed, did you add or modify the demo test page to test the changed/added functionality?
- [x] Did you assign reviewers?
- [x] In Jira, have you linked to this PR on the ticket(s)?

### CODE REVIEW

File: src/scss/components/tabset/_index.scss [hunk 1]
```diff
@@ -137,13 +137,13 @@ hx-tabset:not([hx-defined]) {
 hx-tabset.beta-hxBound {
   hx-tab[current] {
     background-color: $gray-0;
-    background-color: var(--hxTabset-hxTab-current-bgcolor, $gray-0);
+    background-color: var(--hxTabset-hxTab-current-bgcolor, $gray-0);//there is no token for bg color
```
Comment (100stacks) [inline src/scss/components/tabset/_index.scss:140]: Please update to use design tokens. If lower level definition missing, use `$rax-color-gray-0`.

Keep the comment and add a space after the semicolon.

File: src/scss/components/tabset/_index.scss [hunk 2]
```diff
@@ -124,7 +124,7 @@ hx-tabset {
 hx-tabset:not([hx-defined]) {
   hx-tab:first-of-type {
     background-color: $gray-0;
-    background-color: var(--hxTabset-hxTab-first-of-type-bgcolor, $gray-0);
+    background-color: var(--hxTabset-hxTab-first-of-type-bgcolor, $gray-0);//there is no token for bg color
```
Comment (100stacks) [inline src/scss/components/tabset/_index.scss:127]: Please update to use design tokens. If lower level definition missing, use `$rax-color-gray-0`.

Keep the comment and add a space after the semicolon.

File: src/scss/components/tabset/_index.scss [hunk 3]
```diff
@@ -137,13 +137,13 @@ hx-tabset:not([hx-defined]) {
...
   hx-tab[current] {
     background-color: $gray-0;
-    background-color: var(--hxTabset-hxTab-current-bgcolor, $gray-0);
+    background-color: var(--hxTabset-hxTab-current-bgcolor, $gray-0);//there is no token for bg color
   }
 
   hx-tabcontent {
     background-color: $gray-0;
-    background-color: var(--hxTabset-hxTabcontent-bgcolor, $gray-0);
+    background-color: var(--hxTabset-hxTabcontent-bgcolor, $gray-0);//there is no token for bg color
```
Comment (100stacks) [inline src/scss/components/tabset/_index.scss:145]: Please update to use design tokens. If lower level definition missing, use `$rax-color-gray-0`.

Keep the comment and add a space after the semicolon.

File: src/scss/components/tabset/_index.scss [hunk 4]
```diff
@@ -137,13 +137,13 @@ hx-tabset:not([hx-defined]) {
...
     border-color: $gray-300;
-    border-color: var(--hxTabset-hxTabcontent-border-color, $gray-300);
+    border-color: var(--hxTabset-hxTabcontent-border-color, $gray-300);// there is no token for border color
```
Comment (100stacks) [inline src/scss/components/tabset/_index.scss:147]: Please update to use design tokens. If lower level definition missing, use `$rax-color-gray-300`.

Keep the comment and add a space after the semicolon.

File: src/scss/components/tabset/_index.scss [hunk 5]
```diff
@@ -44,8 +44,8 @@ hx-tabcontent {
   background-color: inherit;
   border-color: $gray-300 transparent;
   border-color: var(--hxTabset-hxTabcontent-border-color, $gray-300) transparent;
-  border-style: solid;
-  border-width: 1px;
+  border-style: $rax-component-tab-base-border-style;
+  border-width: $rax-component-tab-base-border-width;//in token border-with is 2px
```
Comment (100stacks) [inline src/scss/components/tabset/_index.scss:48]: Add a space after the semicolon.

**Review by 100stacks — CHANGES_REQUESTED** [review]

File: src/scss/components/tabset/_index.scss [hunk 6]
```diff
@@ -4,15 +4,15 @@
 
 // ===== LAYOUT & APPEARANCE ===============
 hx-tab {
-  border-bottom: 2px solid transparent;
+  border-bottom: $rax-component-tab-base-border-width solid transparent;
   color: $gray-700;
-  color: var(--hxTabset-hxTab-color, $gray-700);
+  color: $rax-component-tab-default-color;
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/tabset/_index.scss:8]: Changes have been done, please review.

File: src/scss/components/tabset/_index.scss [hunk 7]
```diff
@@ -4,15 +4,15 @@
...
   cursor: pointer;
-  font-size: 0.875rem;
+  font-size: $rax-font-size-75;
 
   &:hover {
     color: $gray-950;
-    color: var(--hxTabset-hxTab-hover-color, $gray-950);
+    color: $rax-component-tab-hover-color;
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/tabset/_index.scss:14]: Changes have been done, please review.

File: src/scss/components/tabset/_index.scss [hunk 8]
```diff
@@ -26,15 +26,15 @@ hx-tab {
   &[current] {
     background-color: inherit;
     border-bottom: 2px solid $teal-300;
-    border-bottom: 2px solid var(--hxTabset-hxTab-current-border-bottom, $teal-300);
+    border-bottom: 2px solid $rax-component-tab-active-border-color;// in token $teal-500 is used
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/tabset/_index.scss]: Changes have been done, please review.

File: src/scss/components/tabset/_index.scss [hunk 9]
```diff
@@ -26,15 +26,15 @@ hx-tab {
...
     color: $gray-950;
-    color: var(--hxTabset-hxTab-current-color, $gray-950);
-    font-weight: 600;
+    color: $rax-component-tab-active-color;
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/tabset/_index.scss:30]: Changes have been done, please review.

File: src/scss/components/tabset/_index.scss [hunk 10]
```diff
@@ -44,8 +44,8 @@ hx-tabcontent {
   background-color: inherit;
   border-color: $gray-300 transparent;
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/tabset/_index.scss]: Changes have been done, please review.

File: src/scss/components/tabset/_index.scss (hunk 1 above)
Comment (lalithkarikelli) [reply src/scss/components/tabset/_index.scss]: Changes have been done, please review.

File: src/scss/components/tabset/_index.scss (hunk 2 above)
Comment (lalithkarikelli) [reply src/scss/components/tabset/_index.scss]: Changes have been done, please review.

File: src/scss/components/tabset/_index.scss (hunk 3 above)
Comment (lalithkarikelli) [reply src/scss/components/tabset/_index.scss]: Changes have been done, please review.

File: src/scss/components/tabset/_index.scss (hunk 4 above)
Comment (lalithkarikelli) [reply src/scss/components/tabset/_index.scss]: Changes have been done, please review.

File: src/scss/components/tabset/_index.scss (hunk 5 above)
Comment (lalithkarikelli) [reply src/scss/components/tabset/_index.scss]: Changes have been done, please review.

**Comment by lalithkarikelli** [comment]
Rebased with master and squashed all commits.

**Review by badejayayesubabu — APPROVED** [review]
DEV LGTM

**Review by 100stacks — APPROVED** [review]
Dev LGTM

**Pull request merged**